File: platforms/android/Tatanka/src/de/digitalemil/eagle/Globals.java
package de.digitalemil.eagle;

public class Globals {
	private static int width, height;
	private static float scale;
	private static int w2;
	private static int h2;
	private static Thing[] allThings;
	private static int defaultWidth= 768;
	private static int defaultHeight= 1024;
	protected static int frames= 0;
	
	public static int getFrames() {
		return frames;
	}

	public static int getWidth() {
		return width;
	}
	
	public static int getHeight() {
		return height;
	}
	
	public static int getW2() {
		return w2;
	}
	
	public static int getH2() {
		return h2;
	}
	
	public static float getScale() {
		return scale;
	}
	

	private static void calcScale() { 
		scale = Math.min((float)width / defaultWidth,
				(float)height / defaultHeight);
		System.out.println("scale: "+scale);
	}
	
	private static void setWidth(int width) {
		Globals.width = width;
		w2= width/2;
	}
	
	private static void setHeight(int height) {
		Globals.height = height;
		h2= height/2;
	}
	
	public static void set(int w, int h) {
		setWidth(w);
		setHeight(h);
		calcScale();
	}
	
	protected static void setAllThings(Thing[] all) {
		allThings= all;
	}
	
	public static int getDefaultWidth() {
		return defaultWidth;
	}

	public static int getDefaultHeight() {
		return defaultHeight;
	}

	public static void setDefaults(int defaultWidth, int defaultHeight) {
		Globals.defaultWidth = defaultWidth;
		Globals.defaultHeight = defaultHeight;
		calcScale();
	}

	public static Thing[] getAllThings() {
		return allThings;
	}
}

File: platforms/ios/Tatanka/Tatanka/eagletatanka/src/globals.cpp
#include "all.h"

#include "globals.h"

Globals::~Globals() {

}

int Globals::getMaxThing() {

  return maxthings;
}


int Globals::getFrames() {

  return frames;
}


int Globals::getWidth() {

  return width;
}


int Globals::getHeight() {

  return height;
}


int Globals::getW2() {

  return w2;
}


int Globals::getH2() {

  return h2;
}


float Globals::getScale() {

  return scale;
}


void Globals::calcScale() {

  scale=min(((float)width / defaultWidth),((float)height / defaultHeight));
}


void Globals::setWidth(int width) {

  Globals::width=width;
  w2=width / 2;
}


void Globals::setHeight(int height) {

  Globals::height=height;
  h2=height / 2;
}


void Globals::set(int w, int h) {

  setWidth(w);
  setHeight(h);
  calcScale();
}


void Globals::setAllThings(Thing** all, int n) {

  allThings=all;
  maxthings=n;
}


int Globals::getDefaultWidth() {

  return defaultWidth;
}


int Globals::getDefaultHeight() {

  return defaultHeight;
}


void Globals::setDefaults(int defaultWidth, int defaultHeight) {

  Globals::defaultWidth=defaultWidth;
  Globals::defaultHeight=defaultHeight;
  Screen::screens=(Screen**)new void*[Screen::MAXSCREEN];
  calcScale();
}


Thing** Globals::getAllThings() {

  return allThings;
}


int Globals::width = null;
int Globals::height = null;
float Globals::scale = null;
int Globals::w2 = null;
int Globals::h2 = null;
Thing** Globals::allThings = null;
int Globals::defaultWidth = 768;
int Globals::defaultHeight = 1024;
int Globals::frames = 0;
int Globals::maxthings = null;
